dbms/src/DataStreams/HashJoinProbeBlockInputStream.cpp

```diff
@@ -19,17 +19,16 @@ namespace DB
 {
 HashJoinProbeBlockInputStream::HashJoinProbeBlockInputStream(
     const BlockInputStreamPtr & input,
-    const ExpressionActionsPtr & join_probe_actions_,
+    const JoinPtr & join_,
     const String & req_id)
     : log(Logger::get(req_id))
-    , join_probe_actions(join_probe_actions_)
+    , join(join_)
 {
     children.push_back(input);
 
-    if (!join_probe_actions || join_probe_actions->getActions().size() != 1
-        || join_probe_actions->getActions().back().type != ExpressionAction::Type::JOIN)
+    if (join == nullptr)
```

Comment:
`if (join == nullptr)` is useless here..

Reply:
> `if (join == nullptr)` is useless here..

removed.

Reply:
> `if (join == nullptr)` is useless here..

remove

Reply:
I means that use `RUNTIME_CHECK_MSG(join_ptr)` is ok.
Don't need to use
```
if (join == nullptr)
    RUNTIME_CHECK_MSG(join_ptr)
```

Reply:
> I means that use `RUNTIME_CHECK_MSG(join_ptr)` is ok. Don't need to use
> 
> ```
> if (join == nullptr)
>     RUNTIME_CHECK_MSG(join_ptr)
> ```

I misunderstood.  done.